client/components/app.jsx

```diff
@@ -13,14 +13,31 @@ class App extends React.Component {
   componentDidMount() {
     fetch('/api/grades')
       .then(res => res.json())
-      .then(data => this.setState({ grades: data }))
+      .then(data => {
+        this.setState({ grades: data });
+        this.getAverageGrade();
+      })
       .catch(err => console.error(err));
   }
 
+  getAverageGrade() {
+    const grades = this.state.grades.slice(0);
+    let sum = null;
+    grades.forEach(grade => { sum += grade.grade; });
+
+    let average = sum / grades.length;
+    if (isNaN(average)) {
+      average = 'N/A';
+    } else {
+      average = Math.floor(average);
+    }
+    this.setState({ average: average });
```

Comment:
Since the average grade can be calculated based on state, we don't want to store it in state. This method should only return the average. You can call it in your `render()` method and pass its return value into the `Header`.